src/ts/components/ecomodecontainer.ts

```diff
@@ -0,0 +1,126 @@
+import { MediaType, PlayerAPI, SegmentPlaybackEvent, VideoQuality } from 'bitmovin-player';
+import { i18n } from '../localization/i18n';
+import { Container, ContainerConfig } from './container';
+import { EcoModeToggleButton } from './ecomodetogglebutton';
+import { Label, LabelConfig } from './label';
+import { SettingsPanelItem } from './settingspanelitem';
+
+export class EcoModeContainer extends Container<ContainerConfig> {
+  private ecoModeSavedEmissionsItem: SettingsPanelItem;
+  private ecoModeToggleButtonItem: SettingsPanelItem;
+  private emissionsSavedLabel: Label<LabelConfig>;
+  private savedEmissons = 0;
+  private currentEnergyEmission: number;
+
+  constructor(config: ContainerConfig = {}) {
+    super(config);
+
+    const ecoModeToggleButton = new EcoModeToggleButton();
+    const labelEcoMode = new Label({
+      text: i18n.getLocalizer('ecoMode.title'),
+      for: ecoModeToggleButton.getConfig().id,
+      id: 'ecomodelabel',
+    });
+    this.emissionsSavedLabel = new Label({
+      text: this.savedEmissons.toFixed(4) + ' gCO2',
+      cssClass: 'ui-label-savedEnergy',
+    });
+
+    this.ecoModeToggleButtonItem = new SettingsPanelItem(labelEcoMode, ecoModeToggleButton);
+    this.ecoModeSavedEmissionsItem = new SettingsPanelItem('Saved Emissions', this.emissionsSavedLabel, {
+      hidden: true,
+    });
+
+    this.addComponent(this.ecoModeToggleButtonItem);
+    this.addComponent(this.ecoModeSavedEmissionsItem);
+
+    ecoModeToggleButton.onToggleOn.subscribe(() => {
+      this.ecoModeSavedEmissionsItem.show();
+      this.onActivecallback();
+    });
+
+    ecoModeToggleButton.onToggleOff.subscribe(() => {
+      this.ecoModeSavedEmissionsItem.hide();
+      this.onActivecallback();
+    });
+  }
+
+  private onActivecallback: () => void;
+
+  public setOnActiveChangeCallback(callback: () => void) {
+    this.onActivecallback = callback;
+  }
+
+  configure(player: PlayerAPI): void {
+    player.on(player.exports.PlayerEvent.SegmentPlayback, (segment: SegmentPlaybackEvent) => {

```

Comment:
Nit: Use ES6 import; `player.exports` is `deprecated`

Reply:
[Use ES6 import instead of player.exports](https://github.com/bitmovin/bitmovin-player-ui/pull/571/commits/58d5a9aae07ca334eb87abfb395b8ba1fd82e4f8)

Reply:
For some reason breaks the UI. Reverted:
[Revert "Use ES6 import instead of player.exports"](https://github.com/bitmovin/bitmovin-player-ui/pull/571/commits/774acd3374c1783ab3e4a5a39c7175ac464f6703)